### PR TITLE
👌 IMPROVE: Accomodating mystnb 0.17, moved myst_nb to install_requires

### PR DESCRIPTION
**Pull request by AakashGfude**
- [x] Changes necessary for myst-nb 0.17.  fixes https://github.com/executablebooks/sphinx-jupyterbook-latex/issues/110
- [x]  Have moved myst_nb to install_requires. @mmcky should this be done or we update the doc to have `pip install sphinx-jupyterbook-latex[myst]` instead.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -22,24 +22,22 @@ classifiers =
 [options]
 packages = find:
 install_requires =
-    sphinx>=3,<5
+    myst-nb>=0.13,<0.18
```
Comment (mmcky) [inline setup.cfg:25]: @AakashGfude 👍

Comment (mmcky) [reply setup.cfg:25]: do we require any specific version of `sphinx`?

**Review by mmcky — COMMENTED** [review]
@AakashGfude this is looking good -- just a minor comment re: if we need to track sphinx versions.

**Comment by mmcky** [comment]
- [ ] we should also update the test matrix to python=3.9 I think.

**Comment by chrisjsewell** [comment]
Guys this was specifically moved out of the requirements on #55, so I'm really not sure why you are adding back in

**Comment by AakashGfude** [comment]
@chrisjsewell yes I did see that. But we have imported myst_nb in transforms.py, like https://github.com/executablebooks/sphinx-jupyterbook-latex/blob/master/sphinx_jupyterbook_latex/transforms.py#L486. 
Or we should not run those transforms probably if myst_nb is not installed?

File: setup.cfg (hunk 1 above)
Comment (AakashGfude) [reply setup.cfg:25]: @mmcky  that is being handled by myst-nb, and matches with this extensions requirement.

Comment (mmcky) [reply setup.cfg:25]: Ah - makes sense 👍

**Comment by chrisjsewell** [comment]
It was changed due to circular dependency issues in jupyter-book. It's ok if this is not an issue anymore, but yeh just have a check.

**Comment by mmcky** [comment]
@AakashGfude this will lock us into ensuring releases of `sphinx-jupyterbook-latex` and `jupyter-book` are synchronised to avoid these things. Perhaps it best to:

1. check if `myst-nb` is installed and if so treat the Cells (as per the PR)
2. drop back to a default behaviour if extension is used without `myst-nb`

**Pull request merged**